lk2nd/smp/cpu-boot.c

```diff
@@ -23,6 +23,61 @@
 #define APC_PWR_GATE_CTL_GHDS_EN	BIT(0)
 #define APC_PWR_GATE_CTL_GHDS_CNT(cnt)	((cnt) << 24)
 
+/*
+ * Each CPU within a cluster has a clock controller.
+ * There is just once L2 cache controller per cluster. We can take the base
+ * address of a given CPU clock controller and using a combination of mask
+ * and offset find the base address of the L2 cache controller.
+ *
+ * APCS_GLB_BASE takes the base of given CPU clock controller and offsets
+ * back the single APCS_GLB register which is the base for the L2
+ * cache controller in the cluster.
+ *
+ */
+#define APCS_GLB_MASK			0xfff00000
+#define APCS_GLB_OFFSET			0x00011000
+#define APCS_GLB_BASE(base)		(((base) & APCS_GLB_MASK) + \
+					APCS_GLB_OFFSET)
+
+#define APCS_PWR_CTL_OVERRIDE			0xc
+#define APCS_PWR_CTL_OVR_PRESETDBG		BIT(22)
+
+#define APCS_L2_PWR_CTL				0x14
+#define APCS_L2_PWR_CTL_PMIC_APC_ON		BIT(28)
+#define APCS_L2_PWR_CTL_L2_HS_CNT(cnt)		((cnt) << 16)
+#define APCS_L2_PWR_CTL_L2_ARRAY_HS_CLAMP	BIT(15)
+#define APCS_L2_PWR_CTL_SCU_ARRAY_HS_CLAMP	BIT(14)
+#define APCS_L2_PWR_CTL_L2_RET_SLP		BIT(13)
+#define APCS_L2_PWR_CTL_SYS_RESET		BIT(12)
+#define APCS_L2_PWR_CTL_L2_SLEEP_STATE		BIT(11)
+#define APCS_L2_PWR_CTL_L2_HS_RST		BIT(10)
+#define APCS_L2_PWR_CTL_L2_HS_EN		BIT(9)
+#define APCS_L2_PWR_CTL_L2_HS_CLAMP		BIT(8)
+#define APCS_L2_PWR_CTL_L2_RST_DIS		BIT(2)
+#define APCS_L2_PWR_CTL_SCU_ARRAY_HS		BIT(1)
+#define APCS_L2_PWR_CTL_L2_ARRAY_HS		BIT(0)
+
+#define APCS_L2_PWR_STATUS			0x18
+#define APCS_L2_PWR_STATUS_APC_SUPPLY_ON	BIT(28)
+#define APCS_L2_PWR_STATUS_SAW_SLP_ACK		BIT(27)
+#define APCS_L2_PWR_STATUS_DBG_PWRUP_REQ	BIT(26)
+#define APCS_L2_PWR_STATUS_L2_CLK_IDLE		BIT(25)
+#define APCS_L2_PWR_STATUS_L2_SLV_IDLE		BIT(24)
+#define APCS_L2_PWR_STATUS_L2_HW_EVENT		(BIT(20) | BIT(19) | BIT(18) | \
+						 BIT(17) | BIT(16))
+#define APCS_L2_PWR_STATUS_DBG_RST_STS		BIT(13)
+#define APCS_L2_PWR_STATUS_L2_HS_RST_STS	BIT(12)
+#define APCS_L2_PWR_STATUS_L2_RET_SLP_STS	BIT(10)
+#define APCS_L2_PWR_STATUS_L2_HS_STS		BIT(9)
+#define APCS_L2_PWR_STATUS_L2_CLMP_STS		BIT(8)
+#define APCS_L2_PWR_STATUS_L2_ARRAY_HS_STS	(BIT(3) | BIT(2) | BIT(1) | \
+						 BIT(0))
```

Comment:
I'd drop unused bits for APCS_L2_PWR_STATUS as well, since we're only using APCS_L2_PWR_STATUS_L2_HS_STS

Reply:
NP I can update - we are waiting on @TravMurav to ack anyway